### PR TITLE
[Scripts] Ganache Make and Revert Snapshot

### PR DESCRIPTION
**Pull request by bh2smith**
Such scripts will allow us to address the issue of making E2E tests independent in [dex-services](https://github.com/gnosis/dex-services/issues/161)

**Script Usage/Test Plan**

```
SNAP_ID=$(truffle exec scripts/ganache/makeSnapshot.js | grep '0x')
echo $SNAP_ID
truffle exec scripts/ganache/revertSnapshot.js $SNAP_ID
```

Notice that making the snapshot will require storing part of the output (the snapshot id) which can be used for snapshot revert. Also, since these are truffle scripts, we have to `grep` for the appropriate output.

One can (at least partially) verify that these are working by checking the ganache logs;

Screenshot 2019-06-27 at 11 11 27 AM

### CODE REVIEW

**Review by josojo — APPROVED** [review]
Nice! That will make things easier in the future

File: scripts/ganache/makeSnapshot.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const makeSnapshot = () => {
+  return new Promise((resolve, reject) => {
+    web3.currentProvider.send({
```
Comment (fleupold) [inline scripts/ganache/makeSnapshot.js:3]: We have a nice async wrapper around this send method [here](https://github.com/gnosis/dex-contracts/blob/master/test/utilities.js#L105), which would be more concise to use.

**Review by fleupold — APPROVED** [review]
This is a really useful tool. Since it doesn't depend on anything dƒusion specific it could even be packaged into a npm package (might be worth thinking about it if we need such a script again). It's a bit like git stash.

File: scripts/ganache/makeSnapshot.js (hunk 1 above)
Comment (bh2smith) [reply scripts/ganache/makeSnapshot.js:3]: Do you mean something like `await web3.currentProvider.send(...)`

Comment (fleupold) [reply scripts/ganache/makeSnapshot.js:3]: Something like https://github.com/gnosis/dex-contracts/blob/master/test/utilities.js#L119

Comment (bh2smith) [reply scripts/ganache/makeSnapshot.js:3]: Yes, thanks! I will see how that works!

**Comment by fleupold** [comment]
Are we using this script now? Weren't you planning to do this as a separate npm package?

**Comment by bh2smith** [comment]
Right ya. I was planning to do that. It was an accidental merge. I thought was the multi caller.

**Pull request merged**